util/chunk/row.go

```diff
@@ -189,7 +189,15 @@ func (r Row) GetDatum(colIdx int, tp *types.FieldType) types.Datum {
 		if !r.IsNull(colIdx) {
 			d.SetMysqlDecimal(r.GetMyDecimal(colIdx))
 			d.SetLength(tp.Flen)
-			d.SetFrac(tp.Decimal)
+			// If tp.Decimal is unspecified(-1),
+			// we should set it to the real fraction length of the decimal value,
+			// if not, the d.Frac will be set to MAX_UINT16 which will cause unexpected BadNumber error
+			// when encoding.
```

Comment:
Please make the block comment looks like a "block", you can take "llvm", "impala" or any other projects as examples:
- https://github.com/llvm-mirror/llvm/blob/master/include/llvm/IR/Function.h#L104
- https://github.com/cloudera/Impala/blob/cdh5-trunk/be/src/codegen/llvm-codegen.h#L146